jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,659 @@
+    }
+
+    /**
+     * Include {@link PathSpec}s in the conditions to be met
+     * @param paths String representations of {@link PathSpec}s that are
+     *              tested against the {@link Request#getPathInContext(Request) pathInContext}.
+     */
+    public void includePath(String... paths)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        _paths.include(paths);
+    }
+
+    /**
+     * Exclude {@link PathSpec} in the conditions to be met
+     * @param paths String representations of {@link PathSpec}s that are
+     *              tested against the {@link Request#getPathInContext(Request) pathInContext}.
+     */
+    public void excludePath(String... paths)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        _paths.exclude(paths);
+    }
+
+    /**
+     * Include {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void include(InetAddressPattern... patterns)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (InetAddressPattern p : patterns)
+            _predicates.include(new InetAddressPredicate(p));
+    }
+
+    /**
+     * Include {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void includeInetAddress(String... patterns)
+    {
+        for (String p : patterns)
+            include(InetAddressPattern.from(p));
+    }
+
+    /**
+     * Exclude {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void exclude(InetAddressPattern... patterns)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (InetAddressPattern p : patterns)
+            _predicates.exclude(new InetAddressPredicate(p));
+    }
+
+    /**
+     * Exclude {@link InetAddressPattern} in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void excludeInetAddress(String... patterns)
```

Comment:
This should just be a use of a new inner class InetAddressPredicate with `Predicate.not(new InetAddressPredicate(...))` that is (eventually) composited together with the other Predicates for this `ConditionalHandler`

Reply:
ditto